### DISCUSSION TITLE
Any plans for supporting SQLite?

### DISCUSSION THREAD

**Discussion by timuckun** [patterns-ai-core/langchainrb#770] [General]
The title says it all. Seems like it would be a popular option.

**Comment by andreibondarev** [comment]
@timuckun For RAG/vectorsearch capabilities, right? I had given https://github.com/asg017/sqlite-vss a spin a few months ago but had a lot of issues with it. It could be worth giving https://github.com/asg017/sqlite-vec a try!

**Comment by timuckun** [comment]
I wonder what turso is using.  https://turso.tech/vector

Unfortunately there are no native ruby clients for turso.

**Reply by visini** [reply]
FYI: There is now, see https://github.com/tursodatabase/libsql-ruby

**Comment by andreibondarev** [comment]
@timuckun Do you have a current need for a Langchain::Vectorsearch::SQLite adapter?

**Reply by timuckun** [reply]
I could use it in a project I am thinking about. Happy to test it for you while I work on it.

**Reply by andreibondarev** [reply]
That would be amazing!

**Reply by timuckun** [reply]
I can start on the project this weekend. I'll have to some other code to write first but I'll start a project with Sqlite in rails. You let me know when you are ready to test anything.
